thirdeye/thirdeye-frontend/app/pods/components/modals/entity-mapping-modal/component.js

```diff
@@ -0,0 +1,450 @@
+  /**
+   * Construct the url string based on the passed entities
+   * @param {Array} entities - array of entities
+   */
+  makeUrlString(entities) {
+    const urnStrings = entities
+      .map((entity) => `${entity.toURN}`)
+      .join(',');
+
+    if (!urnStrings.length) {
+      return;
+    }
+
+    return `/rootcause/raw?framework=identity&urns=${urnStrings}`;
+  },
+
+  /**
+   * Fetches related Entities
+   */
+  _fetchRelatedEntities: async function() {
+    const metricUrn = get(this, 'metricUrn');
+
+    const entities = await fetch(`/entityMapping/view/fromURN/${metricUrn}`).then(checkStatus);
+    const url = this.makeUrlString(entities);
+    this.builUrnToId(entities);
+
+    if (!url) {
+      return;
+    }
+    const relatedEntities = await fetch(url).then(checkStatus);
+
+    // merges createBy Props
+    relatedEntities.map((item) => {
+      if (!item.urn) {
+        return;
+      }
+      const { createdBy } = _.find(entities, { toURN: item.urn });
+      item.createdBy = createdBy;
+      return item;
+    });
+
+    set(this, '_relatedEntities', relatedEntities);
+  },
+
+  /**
+   * Fetches new entities with caching and diff checking
+  */
+  didReceiveAttrs() {
+    const {
+      metric,
+      _cachedMetric,
+      showEntityMappingModal
+    } = getProperties(this, 'metric', '_cachedMetric', 'showEntityMappingModal');
+
+    if (showEntityMappingModal && metric && !_.isEqual(metric, _cachedMetric)) {
+      set(this, '_cachedMetric', metric);
+
+      this._fetchRelatedEntities();
+    }
+  },
+
+  /**
+   * Checks if the currently selected entity is already in the mapping
+   */
+  mappingExists: computed(
+    'selectedEntity.label',
+    'relatedEntities.@each.alias',
+    'metric',
+    function() {
+      const {
+        selectedEntity: entity,
+        relatedEntities,
+        metric
+      } = getProperties(this, 'selectedEntity', 'relatedEntities', 'metric');
+
+      return (metric.label === entity.alias) || relatedEntities.some((relatedEntity) => {
+        relatedEntity.label === entity.alias;
+      });
+    }
+  ),
+
+  /**
+   * Helper function that sets an error
+   * @param {String} error - the error message
+   */
+  setError(error) {
+    set(this, 'errorMessage', error);
+  },
+
+  /**
+   * Helper function that clears errors
+  */
+  clearError() {
+    set(this, 'errorMessage', '');
+  },
+
+  /**
+   * Entity Mapping columns to be passed into
+   * ember-models-table
+   */
+  entityColumns: [
+    {
+      propertyName: 'type',
+      title: 'Types',
+      filterWithSelect: true,
+      className: 'te-modal__table-cell te-modal__table-cell--capitalized',
+      predefinedFilterOptions: [
+        'Metric',
+        'Other'
+      ]
+    },
+    {
+      template: 'custom/filterLabel',
+      title: 'Filter value',
+      className: 'te-modal__table-cell',
+      propertyName: 'label',
+      sortedBy: 'label',
+      // custom filter function
+      filterFunction(cell, string, record) {
```

Comment:
@ttbach fyi this is how you pass a custom filter when you're using a custom cell